metis-core/metis-core-service/src/main/java/eu/europeana/metis/core/service/ProxiesService.java

```diff
@@ -395,6 +394,52 @@ public RecordsResponse getListOfFileContentsFromPluginExecution(MetisUserView me
     return new RecordsResponse(records);
   }
 
+  /**
+   * Get a list with record contents from the external resource based on a workflow execution and the predecessor
+   * of the given {@link PluginType}.
+   *
+   * @param metisUserView the user wishing to perform this operation
+   * @param workflowExecutionId the execution identifier of the workflow
+   * @param pluginType the {@link ExecutablePluginType} that is to be located inside the workflow
+   * @param ecloudIds the list of ecloud IDs of the records we wish to obtain
+   * @return the list of records from the external resource
+   * @throws GenericMetisException can be one of:
+   * <ul>
+   * <li>{@link MCSException} if an error occurred while retrieving the records from the external
+   * resource</li>
+   * <li>{@link eu.europeana.metis.exception.UserUnauthorizedException} if the user is not
+   * authorized to perform this task</li>
+   * <li>{@link eu.europeana.metis.core.exceptions.NoWorkflowExecutionFoundException} if no workflow
+   * execution exists for the provided identifier</li>
+   * </ul>
+   */
+  public RecordsResponse getListOfFileContentsFromPredecessorPluginExecution(MetisUserView metisUserView,
+          String workflowExecutionId, ExecutablePluginType pluginType, ListOfIds ecloudIds)
+          throws GenericMetisException {
+
+    // Get the right workflow execution and plugin type.
+    final Pair<WorkflowExecution, AbstractExecutablePlugin> executionAndPlugin = getExecutionAndPlugin(
+            metisUserView, workflowExecutionId, pluginType);
+    if (executionAndPlugin == null) {
+      throw new NoWorkflowExecutionFoundException(String
+              .format("No executable plugin of type %s found for workflowExecution with id: %s",
+                      pluginType.name(), workflowExecutionId));
+    }
+
+    Pair<MetisPlugin, WorkflowExecution> predecessorPlugin =
+            dataEvolutionUtils.getPreviousExecutionAndPlugin(executionAndPlugin.getRight(), executionAndPlugin.getLeft().getDatasetId());
+    AbstractExecutablePlugin predecessorExecutablePlugin = (AbstractExecutablePlugin) predecessorPlugin.getLeft();
```

Comment:
It is possible that there is no predecessor plugin. Perhaps check for `predecessorPlugin==null` and throw a similar exception as the one just above.

Reply:
Added